pkg/communications/component.go

```diff
@@ -25,21 +27,33 @@ type Component interface {
 
 type component struct {
 	keycloakCommunicationsClient KeycloakCommunicationsClient
+	tokenProvider                toolbox.OidcTokenProvider
 	logger                       internal.Logger
 }
 
 // NewComponent returns the communications component.
-func NewComponent(keycloakCommunicationsClient KeycloakCommunicationsClient, logger internal.Logger) Component {
+func NewComponent(keycloakCommunicationsClient KeycloakCommunicationsClient, tokenProvider toolbox.OidcTokenProvider, logger internal.Logger) Component {
```

Comment:
I would suggest to also adapt sendSMS method so we have a coherent behavior regarding token for this component.